integration/apidefinition_test.go

```diff
@@ -0,0 +1,129 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/TykTechnologies/tyk-operator/api/model"
+	"github.com/TykTechnologies/tyk-operator/api/v1alpha1"
+	"github.com/matryer/is"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestApiDefinitionCreate(t *testing.T) {
+	var (
+		apiDefWithJSONValidationName = "apidef-json-validation"
+		apiDefListenPath             = "/validation"
+		defaultVersion               = "Default"
+		errorResponseCode            = 422
+		defaultTimeout               = 1 * time.Minute
+	)
+
+	eps := &model.ExtendedPathsSet{
```

Comment:
shouldn't we isolate this into a subtest so that for example when we add unit tests for each type of extenedPath we can just write another subtests instead of a whole test?